kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsServiceImpl.java

```
@@ -0,0 +1,698 @@
+     * @throws IOException
+     */
+    private boolean processVendorDetails(Date jobRunDate, List<VendorDetail> vendorDetails) throws IOException {
+        FileOutputStream dataFileStream = new FileOutputStream(dataFile);
+        List<String> outputVendorLineList = new ArrayList<String>();
+//        List<VendorDetailExtension> updatedVDEList = new ArrayList<VendorDetailExtension>();
+//        List<Note> noteList = new ArrayList<Note>();
+
+        for (VendorDetail vendorDetail : vendorDetails) {
+            String vendorLine = generateVendorLine(vendorDetail);
+            if (!outputVendorLineList.contains(vendorLine)) {
+                outputVendorLineList.add(vendorLine);
+                dataFileStream.write(vendorLine.getBytes());
+//                VendorDetailExtension updatedVDE = updateVendorDetailExtension(vendorDetail);
+                updateVendor(vendorDetail);
+//                Note note = createNote(vendorDetail);
+//                updatedVDEList.add(updatedVDE);
+//                noteList.add(note);
+            }
+        }
+//        getBusinessObjectService().save(updatedVDEList);
+//        getBusinessObjectService().save(noteList);
+        dataFileStream.flush();
+        dataFileStream.close();
+        writeVendorCountFile(outputVendorLineList.size());
+        KualiInteger listSize = new KualiInteger(outputVendorLineList.size());
+        saveLastRunDate(jobRunDate, listSize);
+        return true;
+    }
+
+    /**
+     * write the file containing number of vendors processed for ecustoms
+     *
+     * @param vendorDetails
+     */
+    private void writeVendorCountFile(int size) throws IOException {
+        FileOutputStream countFileStream = new FileOutputStream(vendorCountFile);
+        String outputMessage = "Vendors added to ecustoms file = " + size;
+        LOG.info(outputMessage);
+        countFileStream.write(outputMessage.getBytes());
+        countFileStream.flush();
+        countFileStream.close();

```

Ok, so throwing exception is good for failing fast, but we still do need to try/finally the file/stream resources.